packages/pancake-uikit/src/widgets/Menu/components/UserMenu/index.tsx

```diff
@@ -0,0 +1,157 @@
+  border: 1px solid ${({ theme }) => theme.colors.cardBorder};
+  border-radius: 16px;
+  padding-bottom: 4px;
+  padding-top: 4px;
+  width: 280px;
+  z-index: 101;
+
+  ${UserMenuItem}:first-child {
+    border-radius: 8px 8px 0 0;
+  }
+
+  ${UserMenuItem}:last-child {
+    border-radius: 0 0 8px 8px;
+  }
+`;
+
+const UserMenu: React.FC<UserMenuProps> = ({
+  account,
+  text,
+  avatarSrc,
+  variant = variants.DEFAULT,
+  children,
+  ...props
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [targetRef, setTargetRef] = useState<HTMLDivElement | null>(null);
+  const [tooltipRef, setTooltipRef] = useState<HTMLDivElement | null>(null);
+  const hideTimeout = useRef<number>();
+  const isHoveringOverTooltip = useRef(false);
+  const accountEllipsis = account ? `${account.substring(0, 2)}...${account.substring(account.length - 4)}` : null;
+  const { styles, attributes } = usePopper(targetRef, tooltipRef, {
+    placement: "bottom-end",
+    modifiers: [{ name: "offset", options: { offset: [0, 12] } }],
+  });
+
+  /**
+   * See "useTooltip"
+   */
+  useEffect(() => {
+    const showTooltip = (evt: MouseEvent | TouchEvent) => {
+      setIsOpen(true);
+
+      if (evt.target === targetRef) {
+        clearTimeout(hideTimeout.current);
+      }
+
+      if (evt.target === tooltipRef) {
+        isHoveringOverTooltip.current = true;
+      }
+    };
+
+    const hideTooltip = (evt: MouseEvent | TouchEvent) => {
+      if (hideTimeout.current) {
+        window.clearTimeout(hideTimeout.current);
+      }
+
+      if (evt.target === tooltipRef) {
+        isHoveringOverTooltip.current = false;
+      }
+
+      if (!isHoveringOverTooltip.current) {
+        hideTimeout.current = window.setTimeout(() => {
+          if (!isHoveringOverTooltip.current) {
+            setIsOpen(false);
+          }
+        }, 100);
+      }
+    };
+
+    if (targetRef) {
+      if (isTouchDevice()) {
+        targetRef.addEventListener("touchstart", showTooltip);
+        targetRef.addEventListener("touchend", hideTooltip);
+      } else {
+        targetRef.addEventListener("mouseenter", showTooltip);
+        targetRef.addEventListener("mouseleave", hideTooltip);
+      }
+    }
+
+    if (tooltipRef) {
+      tooltipRef.addEventListener("mouseenter", showTooltip);
+      tooltipRef.addEventListener("mouseleave", hideTooltip);
+    }
+
+    return () => {
+      if (targetRef) {
+        targetRef.removeEventListener("touchstart", showTooltip);
+        targetRef.removeEventListener("touchend", hideTooltip);
+        targetRef.removeEventListener("mouseenter", showTooltip);
+        targetRef.removeEventListener("mouseleave", hideTooltip);
+      }
+
+      if (tooltipRef) {
+        tooltipRef.removeEventListener("mouseenter", showTooltip);
+        tooltipRef.removeEventListener("mouseleave", hideTooltip);
+      }
+    };
+  }, [targetRef, tooltipRef, hideTimeout, isHoveringOverTooltip, setIsOpen]);
```

Comment:
Very likely that the effect will re rerun without cleaning there.
Put the return in a cleaning function, and call the function when the the effect runs.